kong/plugins/key-auth/handler.lua

```diff
@@ -17,13 +17,20 @@ local KeyAuthHandler = BasePlugin:extend()
 
 KeyAuthHandler.PRIORITY = 1000
 
+local OPTIONS = "OPTIONS"
+
 function KeyAuthHandler:new()
   KeyAuthHandler.super.new(self, "key-auth")
 end
 
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- do not authenticate preflight OPTIONS requests
+  if ngx.req.get_method() == OPTIONS then
+    return
+  end
+
```

Comment:
Considering [this comment and linked question](https://stackoverflow.com/questions/15734031/why-does-the-preflight-options-request-of-an-authenticated-cors-request-work-in/15734032#comment31192566_15734032) I'd prefer a defensive approach; make it an option and default to authenticating `OPTIONS` requests.